### PR TITLE
[Debt]Fix a number of TypeScript errors

### PR DESCRIPTION
**Pull request by gmartin7**
* Make DirectoryWatcher.start() and onChanged() public
* Change all 'bool' to 'boolean' in jquery.d.ts
* 'Fix' MessageEvent problem in synphonyApi.ts
     by calling it from accordion.js

### CODE REVIEW

File: DistFiles/factoryCollections/Templates/Wall Calendar/configure.js [hunk 1]
```diff
@@ -144,4 +133,4 @@ var TestCalendar = (function () {
     }
     return TestCalendar;
 })();
-//# sourceMappingURL=configure.js.map
+//# sourceMappingURL=configure.js.map
```
Comment (StephenMcConnel) [inline DistFiles/factoryCollections/Templates/Wall Calendar/configure.js]: Did you remove the newline at the end of the file?  If so, was it deliberate?

File: src/BloomBrowserUI/bookEdit/js/directoryWatcher.js [hunk 2]
```diff
@@ -66,41 +59,37 @@ var DirectoryWatcher = (function () {
...
+     * @param listenerNameAndContext Name and context that identifies the handler to remove.
+     */
     DirectoryWatcher.prototype.offChanged = function (listenerNameAndContext) {
         if (this.changeEventHandlers.hasOwnProperty(listenerNameAndContext))
             delete this.changeEventHandlers[listenerNameAndContext];
     };
     return DirectoryWatcher;
 })();
-//# sourceMappingURL=directoryWatcher.js.map
+//# sourceMappingURL=directoryWatcher.js.map
```
Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/directoryWatcher.js]: Same questions about newline at end of the file.

File: src/BloomBrowserUI/bookEdit/js/synphonyApi.js [hunk 3]
```diff
@@ -44,78 +38,64 @@ var SynphonyApi = (function () {
...
         return this.levels;
     };
-
     SynphonyApi.prototype.addLevel = function (aLevel) {
         this.levels.push(aLevel);
     };
     return SynphonyApi;
 })();
-//# sourceMappingURL=synphonyApi.js.map
+//# sourceMappingURL=synphonyApi.js.map
```
Comment (StephenMcConnel) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: Same questions for newline and end of file.

**Comment by StephenMcConnel** [comment]
Generally LGTM, but did you intend to remove newlines from three of the files? (or did you -- I think the diff shows the absence of a newline is new.)

**Comment by gmartin7** [comment]
I didn't do anything special to the end of the files. These are .js files that are generated from .ts files.

**Comment by StephenMcConnel** [comment]
OK.  :+1:

File: DistFiles/factoryCollections/Templates/Wall Calendar/configure.ts [hunk 4]
```diff
@@ -43,11 +43,11 @@ class CalendarConfigurator {
...
     updateDom():void {
         var year:string = this.configObject.year;
-        var originalMonthsPicturePage:HTMLDivElement = $('.calendarMonthTop')[0];
-        var pageToInsertAfter:HTMLDivElement = originalMonthsPicturePage;
+        var originalMonthsPicturePage:HTMLElement = $('.calendarMonthTop')[0];
+        var pageToInsertAfter:HTMLElement = originalMonthsPicturePage;
         for (var month:number = 0; month < 12; month++) {
 
-            var monthsPicturePage:HTMLDivElement = $(originalMonthsPicturePage).clone();
+            var monthsPicturePage:HTMLElement = $(originalMonthsPicturePage).clone()[0];
```
Comment (JohnThomson) [inline DistFiles/factoryCollections/Templates/Wall Calendar/configure.ts]: Adding the [0] here feels like a substantive change. Probably good, but I had the impression you were just fixing up spurious warnings.

File: src/BloomBrowserUI/bookEdit/js/synphonyApi.js [hunk 5]
```diff
@@ -44,78 +38,64 @@ var SynphonyApi = (function () {
...
-
             this.source.letters = sorted.join(' ');
         }
     };
-
     SynphonyApi.fireCSharpEvent = function (eventName, eventData) {
-        //noinspection TaskProblemsInspection
-        var event = new MessageEvent(eventName, { 'view': window, 'bubbles': true, 'cancelable': true, 'data': eventData });
-        document.dispatchEvent(event);
+        fireCSharpAccordionEvent(eventName, eventData);
```
Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: Again, this feels like a substantive change not obviously related to the task at hand.

File: DistFiles/factoryCollections/Templates/Wall Calendar/configure.ts (hunk 4 above)
Comment (gmartin7) [inline DistFiles/factoryCollections/Templates/Wall Calendar/configure.ts]: Clone() returned a JQuery object, not an HTMLElement (or HTMLDivElement). In order to fix the error messages here caused by type-checking, I had to specify the internal HTMLElement in the JQuery object. It was an interesting 'feature' of JQuery that it actually worked before! Fortunately, the jquery object always contained one and only one item, which was the page we wanted.

File: src/BloomBrowserUI/bookEdit/js/synphonyApi.js (hunk 5 above)
Comment (gmartin7) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: The MessageEvent ctor works fine in javascript, but it's actually not declared anywhere in TypeScript-land, so it generates an error when called from TypeScript. Phillip and I both couldn't figure out how to tell TypeScript it was okay. Then I found that accordion.js (not generated from TS) had defined an identical function and was loaded and available. Calling that js function here avoids the TS issue and makes us only have the function in one place.

Comment (hatton) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: Calling a  function in UI code, from a function in model code, that talks to more UI code somewhere... this is smelling really bad.

Comment (gmartin7) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: These 'fireCSharpEvent' methods appear to be scattered all over. Would it be any better if I used the one in bloomEditing.js? Unless someone can figure out how to get Typescript to accept this version of a MessageEvent ctor, we need the version of the method that we use to be in un-generated javascript.

Comment (JohnThomson) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: According to Resharper, MessageEvent is defined (in lib.d.ts), but as an interface, not a class. I don't know typescript well, but creating an instance of something declared only as an interface would not be allowed at all in C#; it feels weird.

Comment (phillip-hopper) [inline src/BloomBrowserUI/bookEdit/js/synphonyApi.js]: The *.d.ts files are not actual code files, they're more like the metadata files you get sometimes when in C# you go to the declaration of a function for which no code is available. They are used to help you get the number and type of parameters correct, and to provide code completion lists.

**Comment by hatton** [comment]
@JohnThomson at one point this was waiting on me to investigate something. That was resolved. Is there anything else you need Gordon to do?

**Pull request merged**